internal/versionsapi/imageinfo.go

```diff
@@ -102,15 +91,51 @@ func (i ImageInfo) Validate() error {
 	if !semver.IsValid(i.Version) {
 		retErr = errors.Join(retErr, fmt.Errorf("version %q is not a valid semver", i.Version))
 	}
-	var providers int
-	providers += len(i.AWS)
-	providers += len(i.Azure)
-	providers += len(i.GCP)
-	providers += len(i.OpenStack)
-	providers += len(i.QEMU)
-	if providers == 0 {
-		retErr = errors.Join(retErr, errors.New("one or more providers must be specified"))
+	if len(i.List) == 0 {
+		retErr = errors.Join(retErr, errors.New("one or more image variants must be specified in the list"))
 	}
 
 	return retErr
 }
+
+// MergeImageInfos combines the image info entries from multiple sources into a single
+// image info object.
+func MergeImageInfos(infos ...ImageInfo) (ImageInfo, error) {
```

Comment:
This could be merged with the measurements merging func using generics, not sure if its worth the effort tho